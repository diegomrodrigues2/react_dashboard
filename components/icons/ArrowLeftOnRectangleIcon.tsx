import React from 'react';
import IconBase, { IconBaseProps } from './IconBase';

// Heroicons: arrow-left-on-rectangle (for Logout)
const ArrowLeftOnRectangleIcon: React.FC<IconBaseProps> = (props) => (
  <IconBase {...props}>
    <path strokeLinecap="round" strokeLinejoin="round" d="M15.75 9V5.25A2.25 2.25 0 0 0 13.5 3h-6a2.25 2.25 0 0 0-2.25 2.25v13.5A2.25 2.25 0 0 0 7.5 21h6a2.25 2.25 0 0 0 2.25-2.25V15m-3 0-3-3m0 0 3-3m-3 3H15.75" />
  </IconBase>
);

export default ArrowLeftOnRectangleIcon;
