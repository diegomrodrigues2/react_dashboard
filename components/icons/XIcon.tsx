import React from 'react';
import IconBase, { IconBaseProps } from './IconBase';

const XIcon: React.FC<IconBaseProps> = (props) => (
  <IconBase {...props}>
    <path strokeLinecap="round" strokeLinejoin="round" d="M6 18L18 6M6 6l12 12" />
  </IconBase>
);

export default XIcon;
