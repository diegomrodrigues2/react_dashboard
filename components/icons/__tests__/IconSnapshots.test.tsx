import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import ArrowLeftOnRectangleIcon from '../ArrowLeftOnRectangleIcon';
import ChartPieIcon from '../ChartPieIcon';
import XIcon from '../XIcon';

describe('Icon snapshots', () => {
  it('ArrowLeftOnRectangleIcon matches snapshot', () => {
    const { container } = render(<ArrowLeftOnRectangleIcon aria-label="logout" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('ChartPieIcon matches snapshot', () => {
    const { container } = render(<ChartPieIcon aria-label="chart" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('XIcon matches snapshot', () => {
    const { container } = render(<XIcon aria-label="close" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
