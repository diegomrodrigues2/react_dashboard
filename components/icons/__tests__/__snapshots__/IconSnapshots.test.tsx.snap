// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon snapshots > ArrowLeftOnRectangleIcon matches snapshot 1`] = `
<svg
  aria-label="logout"
  fill="none"
  stroke="currentColor"
  stroke-width="1.5"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M15.75 9V5.25A2.25 2.25 0 0 0 13.5 3h-6a2.25 2.25 0 0 0-2.25 2.25v13.5A2.25 2.25 0 0 0 7.5 21h6a2.25 2.25 0 0 0 2.25-2.25V15m-3 0-3-3m0 0 3-3m-3 3H15.75"
    stroke-linecap="round"
    stroke-linejoin="round"
  />
</svg>
`;

exports[`Icon snapshots > ChartPieIcon matches snapshot 1`] = `
<svg
  aria-label="chart"
  fill="none"
  stroke="currentColor"
  stroke-width="1.5"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M10.5 6a7.5 7.5 0 107.5 7.5h-7.5V6z"
    stroke-linecap="round"
    stroke-linejoin="round"
  />
  <path
    d="M13.5 10.5H21A7.5 7.5 0 0013.5 3v7.5z"
    stroke-linecap="round"
    stroke-linejoin="round"
  />
</svg>
`;

exports[`Icon snapshots > XIcon matches snapshot 1`] = `
<svg
  aria-label="close"
  fill="none"
  stroke="currentColor"
  stroke-width="1.5"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M6 18L18 6M6 6l12 12"
    stroke-linecap="round"
    stroke-linejoin="round"
  />
</svg>
`;
